Makefile

```diff
@@ -0,0 +1,40 @@
+# define the name of the virtual environment directory
+VENV := venv
+
+# default target, when make executed without arguments
+all: venv
+
+install: FORCE
+	pip install --upgrade pip
+	pip install -r requirements.txt --require-venv
+	pip install -r requirements-optional.txt --require-venv
+	pip install --editable . --require-venv
+
+requirements.txt:
+	pip install -r requirements.txt --require-venv
+
+# update is a shortcut target
+update: requirements.txt
+
+test: update
+	# Executes pytests which are located in tests/
+	pytest
+	# Executes filecheck tests
+	lit tests/filecheck
+
+format:
+	yapf -ir xdsl
+
+lint:
+	pyright
+
+lint-ci: pyright-ci.json
+	pyright -p pyright-ci.json
+
+clean:
+	rm -r $(VENV)
+	find . -type f -name '*.pyc' -delete
+
+.PHONY: all install update venv test clean
```

Comment:
`venv` should probably not be phony, right? Am I missing something?

Reply:
To be honest I'm winging it with a fair bit of this makefile, would appreciate help on how to improve it. I copied an existing makefile from a tutorial and have just been adding the rules to the .PHONY line

Reply:
If you want we can try this together, I'm also not super familiar with how Makefiles work, but I'm sure we can figure out a nice way together!